plugins/modules/support_levels.py

```diff
@@ -119,18 +147,19 @@ def run_module():
         'Authorization': f'Bearer {token}'
     }
 
+    resource_type = module.params.get('resource_type')
     query_params = {}
 
     for k in QUERY_PARAMS_LIST:
         val = module.params.get(k)
         if val:
             query_params = query_params | {k: val}
 
-    response = requests.get(f"{API_URL}/support-levels/features", params=query_params, headers=headers)
+    response = requests.get(f"{API_URL}/support-levels/{resource_type}", params=query_params, headers=headers)
```

Comment:
We may need to create two different QUERY_PARAMS_LIST as they are different for architectures and features, right?

Reply:
The /v2/support-levels/architectures endpoint only recognizes 'openshift_version'  as a query parameter. However, I tested with additional parameters and it ignored them, so having a single QUERY_PARAMS_LIST doesn't impact module functionality, though the intent might be clearer if they were separated.

Reply:
I agree that the intent would be clear if separated and easy to read.

Reply:
@vjayaramrh I separated QUERY_PARAMS_LIST based on resource_type in the latest commit.